app/scripts/modules/amazon/serverGroup/details/serverGroupDetails.aws.controller.js

```diff
@@ -418,5 +425,46 @@ module.exports = angular.module('spinnaker.serverGroup.details.aws.controller',
         serverGroup.accountDetails = details;
       });
     };
+
+    this.lookForPipelineCommitDiffs = (executionId) => {
+      if (executionId) {
+        executionService.getExecution(executionId).then(executionDetails => {
+
+          this.jarDiffs = {
+            added: [],
+            downgraded: [],
+            duplicates: [],
+            removed: [],
+            unchanged: [],
+            unknown: [],
+            upgraded: []
+          };
+          this.commits = [];
+
+          const deployStages = executionDetails.stages.filter(stage => stage.type === 'deploy');
```

Comment:
The thought here is that pipelines can have multiple deploy stages so the code below here essentially aggregates the data from the multiple stages.  If that logic is not right, PLMK.